Chore: Get local tests passing on all platforms.

## Mac

* [x] Mac + `ROWDY_SETTINGS="local.safari" npm test`
    * Error below: https://github.com/FormidableLabs/rowdy/issues/9#issuecomment-90904911
    * http://elementalselenium.com/tips/69-safari
    * https://code.google.com/p/selenium/issues/detail?id=7933
    * Need manual installation of driver documented in https://github.com/FormidableLabs/rowdy/commit/487174f27b29f9ba5726654ef38905b002aa46c3
* [x] Mac + `ROWDY_SETTINGS="local.phantomjs" npm test`
* [x] Mac + `ROWDY_SETTINGS="local.firefox" npm test`
* [x] Mac + `ROWDY_SETTINGS="local.chrome" npm test`

## Windows

* [x] Win + `cmd /C "set ROWDY_SETTINGS=local.ie && npm test"`
    * Error below: https://github.com/FormidableLabs/rowdy/issues/9#issuecomment-90903336
    * Need manual Windows tweaking now documented in https://github.com/FormidableLabs/rowdy/commit/0d8eaee5fff670b07304a3e4cd8a8a8f9f12a552
* [x] Win + `cmd /C "set ROWDY_SETTINGS=local.phantomjs && npm test"`
    * Needs fix for https://github.com/FormidableLabs/rowdy/issues/10 in `chore-win-compat`
* [x] Win + `cmd /C "set ROWDY_SETTINGS=local.firefox && npm test"`
* [x] Win + `cmd /C "set ROWDY_SETTINGS=local.chrome && npm test"`

## Win + IE Failure

```
X:\scm\fmd\rowdy>cmd /C "set ROWDY_SETTINGS=local.ie && npm test"

> rowdy@0.0.5 test X:\scm\fmd\rowdy
> cd examples/mocha && mocha

child_process: customFds option is deprecated, use stdio instead.


[SERVER] [stderr] 05:43:54.957 INFO - Launching a standalone server
[SERVER] [stderr] Setting system property webdriver.chrome.driver to X:\scm\fmd\rowdy\node_modules\selenium-standalone\.selenium\chromedriver\2.14-ia32-chromedriver
[SERVER] [stderr] Setting system property webdriver.ie.driver to X:\scm\fmd\rowdy\node_modules\selenium-standalone\.selenium\iedriver\2.45.0-ia32-IEDriverServer.exe
Setting system property phantomjs.binary.path to ..\..\node_modules\phantomjs\lib\phantom\phantomjs.exe
[SERVER] [stderr] 05:43:55.096 INFO - Java: Oracle Corporation 25.40-b25
05:43:55.096 INFO - OS: Windows 7 6.1 x86
[SERVER] [stderr] 05:43:55.127 INFO - v2.45.0, with Core v2.45.0. Built from revision 5017cb8
[SERVER] [stderr] 05:43:55.421 INFO - RemoteWebDriver instances should connect to: http://127.0.0.1:4444/wd/hub
[SERVER] [stderr] 05:43:55.436 INFO - Version Jetty/5.1.x
[SERVER] [stderr] 05:43:55.436 INFO - Started HttpContext[/selenium-server,/selenium-server]
[SERVER] [stderr] 05:43:55.514 INFO - Started org.openqa.jetty.jetty.servlet.ServletHandler@1223dd8
[SERVER] [stderr] 05:43:55.514 INFO - Started HttpContext[/wd,/wd]
05:43:55.514 INFO - Started HttpContext[/selenium-server/driver,/selenium-server/driver]
05:43:55.514 INFO - Started HttpContext[/,/]
[SERVER] [stderr] 05:43:55.514 INFO - Started SocketListener on 0.0.0.0:4444
[CLIENT] [cmd] CALL init({"browserName":"internet explorer"})
[CLIENT] [http] POST /session {"desiredCapabilities":{"browserName":"internet explorer","version":"","javascriptEnabled":true,"platform":"ANY"}}
[SERVER] [stderr] 05:43:55.514 INFO - Started org.openqa.jetty.jetty.Server@133314b
[SERVER] [stderr] 05:43:56.086 INFO - Executing: [new session: Capabilities [{browserName=internet explorer, javascriptEnabled=true, version=, platform=ANY}]])
[SERVER] [stderr] 05:43:56.117 INFO - Creating a new session for Capabilities [{browserName=internet explorer, javascriptEnabled=true, version=, platform=ANY}]
[SERVER] [stderr] Started InternetExplorerDriver server (32-bit)
2.45.0.0
Listening on port 18260
[SERVER] [stderr] 05:43:59.347 INFO - Done: [new session: Capabilities [{browserName=internet explorer, javascriptEnabled=true, version=, platform=ANY}]]
[CLIENT] [status] Driving the web on session: 33d2b7af-39e9-4334-bd06-db551c4e6ba7
[CLIENT] [cmd] RESPONSE init({"browserName":"internet explorer"}) "33d2b7af-39e9-4334-bd06-db551c4e6ba7",{"browserAttachTimeout":0,"enablePersistentHover":true,"ie.forceCreateProcessApi":false,"ie.usePerProcessProxy":false,"ignoreZoomSetting":false,"handlesAlerts":true,"version":"11","platform":"WINDOWS","nativeEvents":true,"ie.ensureCleanSession":false,"elementScrollBehavior":0,"ie.browserCommandLineSwitches":"","webdriver.remote.sessionid":"33d2b7af-39e9-4334-bd06-db551c4e6ba7","requireWindowFocus":false,"browserName":"internet explorer","initialBrowserUrl":"http://localhost:18260/","takesScreenshot":true,"javascriptEnabled":true,"ignoreProtectedModeSettings":false,"enableElementCacheCleanup":true,"cssSelectorsEnabled":true,"unexpectedAlertBehaviour":"dismiss"}
[CLIENT] [cmd] CALL setImplicitWaitTimeout(200)
[CLIENT] [http] POST /session/:sessionID/timeouts/implicit_wait {"ms":200}
[SERVER] [stderr] 05:43:59.381 INFO - Executing: [implicitly wait: 200])
[SERVER] [stderr] 05:43:59.415 INFO - Done: [implicitly wait: 200]
[CLIENT] [cmd] RESPONSE setImplicitWaitTimeout(200)
[CLIENT] [cmd] CALL get("http://backbone-testing.com/notes/app/")
[CLIENT] [http] POST /session/:sessionID/url {"url":"http://backbone-testing.com/notes/app/"}
[SERVER] [stderr] 05:43:59.432 INFO - Executing: [get: http://backbone-testing.com/notes/app/])
[SERVER] [stderr] 05:43:59.687 INFO - Done: [get: http://backbone-testing.com/notes/app/]
[CLIENT] [cmd] RESPONSE get("http://backbone-testing.com/notes/app/")
[CLIENT] [cmd] CALL clearLocalStorage()
[CLIENT] [http] POST /session/:sessionID/execute {"script":"/* jshint evil:true
*/var args = Array.prototype.slice.call(arguments, 0);var code = args[0], fargs
= args[1];var wrap = function() {  return eval(code);};return wrap.apply(this, fargs);","args":["localStorage.clear()",[]]}
[SERVER] [stderr] 05:43:59.704 INFO - Executing: [execute script: /* jshint evil:true */var args = Array.prototype.slice.call(arguments, 0);var code = args[0],
fargs = args[1];var wrap = function() {  return eval(code);};return wrap.apply(this, fargs);, [localStorage.clear(), []]])
[SERVER] [stderr] 05:44:00.249,- Exception thrown
org.openqa.selenium.NoSuchWindowException: Unable to get browser (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 17 milliseconds
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'IE9Win7', ip: '10.0.2.15', os.name: 'Windows 7', os.arch: 'x86', os.version: '6.1', java.version: '1.8.0_40'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities [{browserAttachTimeout=0, enablePersistentHover=true, ie.forceCreateProcessApi=false, ie.usePerProcessProxy=false, ignoreZoomSetting=false, handlesAlerts=true, version=11, platform=WINDOWS, nativeEvents=true, ie.ensureCleanSession=false, elementScrollBehavior=0, ie.browserCommandLineSwitches=, requireWindowFocus=false, browserName=internet explorer, initialBrowserUrl=http://localhost:18260/, takesScreenshot=true, javascriptEnabled=true, ignoreProtectedModeSettings=false, enableElementCacheCleanup=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=dismiss}]
Session ID: 4b191fce-1959-459e-bd36-e31a9e08a9e3
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:204)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:156)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599)
        at org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:508)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.openqa.selenium.support.events.EventFiringWebDriver$2.invoke(EventFiringWebDriver.java:101)
        at com.sun.proxy.$Proxy1.executeScript(Unknown Source)
        at org.openqa.selenium.support.events.EventFiringWebDriver.executeScript(EventFiringWebDriver.java:213)
        at org.openqa.selenium.remote.server.handler.ExecuteScript.call(ExecuteScript.java:53)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:168)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
[SERVER] [stderr] 05:44:00.283,- Exception: Unable to get browser (WARNING: The
server did not provide any stacktrace information)
Command duration or timeout: 17 milliseconds
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'IE9Win7', ip: '10.0.2.15', os.name: 'Windows 7', os.arch: 'x86', os.version: '6.1', java.version: '1.8.0_40'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities [{browserAttachTimeout=0, enablePersistentHover=true, ie.forceCreateProcessApi=false, ie.usePerProcessProxy=false, ignoreZoomSetting=false, handlesAlerts=true, version=11, platform=WINDOWS, nativeEvents=true, ie.ensureCleanSession=false, elementScrollBehavior=0, ie.browserCommandLineSwitches=, requireWindowFocus=false, browserName=internet explorer, initialBrowserUrl=http://localhost:18260/, takesScreenshot=true, javascriptEnabled=true, ignoreProtectedModeSettings=false, enableElementCacheCleanup=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=dismiss}]
Session ID: 4b191fce-1959-459e-bd36-e31a9e08a9e3
  1) "before all" hook
[CLIENT] [cmd] CALL quit()
[CLIENT] [http] DELETE /session/:sessionID
[SERVER] [stderr] 05:44:00.334 INFO - Executing: [delete session: 33d2b7af-39e9-4334-bd06-db551c4e6ba7])
[SERVER] [stderr] 05:44:01.508 INFO - Done: [delete session: 33d2b7af-39e9-4334-bd06-db551c4e6ba7]
[CLIENT] [status] Ending your web drivage..
[CLIENT] [cmd] RESPONSE quit()

  0 passing (6s)
  1 failing

  1)  "before all" hook:
     Error: [clearLocalStorage()] Error response status: 23, , NoSuchWindow - A
request to switch to a different window could not be satisfied because the window could not be found. Selenium error: Unable to get browser (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 17 milliseconds
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'IE9Win7', ip: '10.0.2.15', os.name: 'Windows 7', os.arch: 'x86', os.version: '6.1', java.version: '1.8.0_40'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities [{browserAttachTimeout=0, enablePersistentHover=true, ie.forceCreateProcessApi=false, ie.usePerProcessProxy=false, ignoreZoomSetting=false, handlesAlerts=true, version=11, platform=WINDOWS, nativeEvents=true, ie.ensureCleanSession=false, elementScrollBehavior=0, ie.browserCommandLineSwitches=, requireWindowFocus=false, browserName=internet explorer, initialBrowserUrl=http://localhost:18260/, takesScreenshot=true, javascriptEnabled=true, ignoreProtectedModeSettings=false, enableElementCacheCleanup=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=dismiss}]
Session ID: 4b191fce-1959-459e-bd36-e31a9e08a9e3
      at exports.newError (X:\scm\fmd\rowdy\node_modules\wd\lib\utils.js:139:13)
      at X:\scm\fmd\rowdy\node_modules\wd\lib\callbacks.js:75:19
      at X:\scm\fmd\rowdy\node_modules\wd\lib\webdriver.js:169:5
      at Request._callback (X:\scm\fmd\rowdy\node_modules\wd\lib\http-utils.js:84:7)
      at Request.self.callback (X:\scm\fmd\rowdy\node_modules\wd\node_modules\request\request.js:236:22)
      at Request.emit (events.js:110:17)
      at Request.<anonymous> (X:\scm\fmd\rowdy\node_modules\wd\node_modules\request\request.js:1142:14)
      at Request.emit (events.js:129:20)
      at IncomingMessage.<anonymous> (X:\scm\fmd\rowdy\node_modules\wd\node_modules\request\request.js:1096:12)
      at IncomingMessage.emit (events.js:129:20)
      at _stream_readable.js:908:16
      at process._tickCallback (node.js:355:11)



npm ERR! Test failed.  See above for more details.
```

## Mac + Safari Failure

Summary:

```
Exception: Failed to connect to SafariDriver after 10057 ms
Error: [init({"browserName":"safari"})] The environment you requested was unavailable.
```

Opened Browser Window

```
SafariDriver Launcher

[  0.004s] [safaridriver.client] Connecting to SafariDriver browser extension...
[  0.024s] [safaridriver.client] This will fail if you have not installed the latest SafariDriver extension from
http://selenium-release.storage.googleapis.com/index.html
[  0.025s] [safaridriver.client] Extension logs may be viewed by clicking the Selenium [✓] button on the Safari toolbar
[  2.551s] [safaridriver.client] Unable to establish a connection with the SafariDriver extension
```

Full Log:

```
$ ROWDY_OPTIONS='{ "clientLogger": true, "serverLogger": true }'   ROWDY_SETTINGS="local.safari"   npm test

> rowdy@0.0.5 test /Users/rroemer/scm/fmd/rowdy
> cd examples/mocha && mocha



[SERVER] [stderr] 05:49:36.564 INFO - Launching a standalone server
[SERVER] [stderr] Setting system property webdriver.chrome.driver to /Users/rroemer/scm/fmd/rowdy/node_modules/selenium-standalone/.selenium/chromedriver/2.14-x64-chromedriver
[SERVER] [stderr] 
[SERVER] [stderr] 05:49:36.597 INFO - Java: Oracle Corporation 24.60-b09
[SERVER] [stderr] 05:49:36.598 INFO - OS: Mac OS X 10.9.5 x86_64
[SERVER] [stderr] 05:49:36.607 INFO - v2.45.0, with Core v2.45.0. Built from revision 5017cb8
[SERVER] [stderr] 05:49:36.678 INFO - Default driver org.openqa.selenium.ie.InternetExplorerDriver registration is skipped: registration capabilities Capabilities [{platform=WINDOWS, ensureCleanSession=true, browserName=internet explorer, version=}] does not match with current platform: MAC
[SERVER] [stderr] 05:49:36.708 INFO - RemoteWebDriver instances should connect to: http://127.0.0.1:4444/wd/hub
[SERVER] [stderr] 05:49:36.708 INFO - Version Jetty/5.1.x
[SERVER] [stderr] 05:49:36.709 INFO - Started HttpContext[/selenium-server/driver,/selenium-server/driver]
[SERVER] [stderr] 05:49:36.709 INFO - Started HttpContext[/selenium-server,/selenium-server]
[SERVER] [stderr] 05:49:36.710 INFO - Started HttpContext[/,/]
[SERVER] [stderr] 05:49:36.728 INFO - Started org.openqa.jetty.jetty.servlet.ServletHandler@4ce868c9
[SERVER] [stderr] 05:49:36.728 INFO - Started HttpContext[/wd,/wd]
[SERVER] [stderr] 05:49:36.732 INFO - Started SocketListener on 0.0.0.0:4444
[CLIENT] [cmd] CALL init({"browserName":"safari"}) 
[CLIENT] [http] POST /session {"desiredCapabilities":{"browserName":"safari","version":"","javascriptEnabled":true,"platform":"ANY"}}
[SERVER] [stderr] 05:49:36.732 INFO - Started org.openqa.jetty.jetty.Server@67e7b36d
[SERVER] [stderr] 05:49:36.893 INFO - Executing: [new session: Capabilities [{platform=ANY, javascriptEnabled=true, browserName=safari, version=}]])
[SERVER] [stderr] 05:49:36.904 INFO - Creating a new session for Capabilities [{platform=ANY, javascriptEnabled=true, browserName=safari, version=}]
[SERVER] [stderr] 05:49:36.966 INFO - Server started on port 24620
[SERVER] [stderr] 05:49:36.980 INFO - Launching Safari
[SERVER] [stderr] 05:49:36.990 INFO - Waiting for SafariDriver to connect
[SERVER] [stderr] 05:49:46.991 INFO - Shutting down
[SERVER] [stderr] 05:49:46.991 INFO - Stopping Safari
[SERVER] [stderr] 05:49:47.042 INFO - Stopping server
[SERVER] [stderr] 05:49:47.042 INFO - Stopping server
[SERVER] [stderr] 05:49:47.046 INFO - Shutdown complete
[SERVER] [stderr] 05:49:47.047 INFO - Shutting down
[SERVER] [stderr] 05:49:47.047 INFO - Stopping server
05:49:47.047 INFO - Shutdown complete
[SERVER] [stderr] 05:49:47.049,- Exception thrown
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'L-SB80X4FFT4-M.local', ip: '192.168.2.34', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_60'
Driver info: driver.version: unknown
  at java.util.concurrent.FutureTask.report(FutureTask.java:122)
  at java.util.concurrent.FutureTask.get(FutureTask.java:188)
  at org.openqa.selenium.remote.server.DefaultSession.execute(DefaultSession.java:175)
  at org.openqa.selenium.remote.server.DefaultSession.<init>(DefaultSession.java:111)
  at org.openqa.selenium.remote.server.DefaultSession.createSession(DefaultSession.java:88)
  at org.openqa.selenium.remote.server.DefaultDriverSessions.newSession(DefaultDriverSessions.java:109)
  at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:57)
  at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:1)
  at org.openqa.selenium.remote.server.rest.ResultConfig.handle(ResultConfig.java:110)
  at org.openqa.selenium.remote.server.JsonHttpCommandHandler.handleRequest(JsonHttpCommandHandler.java:172)
  at org.openqa.selenium.remote.server.DriverServlet.handleRequest(DriverServlet.java:201)
  at org.openqa.selenium.remote.server.DriverServlet.doPost(DriverServlet.java:163)
  at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
  at org.openqa.selenium.remote.server.DriverServlet.service(DriverServlet.java:129)
  at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
  at org.openqa.jetty.jetty.servlet.ServletHolder.handle(ServletHolder.java:428)
  at org.openqa.jetty.jetty.servlet.ServletHandler.dispatch(ServletHandler.java:680)
  at org.openqa.jetty.jetty.servlet.ServletHandler.handle(ServletHandler.java:571)
  at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1526)
  at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1479)
  at org.openqa.jetty.http.HttpServer.service(HttpServer.java:920)
  at org.openqa.jetty.http.HttpConnection.service(HttpConnection.java:820)
  at org.openqa.jetty.http.HttpConnection.handleNext(HttpConnection.java:986)
  at org.openqa.jetty.http.HttpConnection.handle(HttpConnection.java:837)
  at org.openqa.jetty.http.SocketListener.handleConnection(SocketListener.java:243)
  at org.openqa.jetty.util.ThreadedServer.handle(ThreadedServer.java:358)
  at org.openqa.jetty.util.ThreadPool$PoolThread.run(ThreadPool.java:537)
Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'L-SB80X4FFT4-M.local', ip: '192.168.2.34', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_60'
Driver info: driver.version: unknown
  at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:69)
  at org.openqa.selenium.remote.server.DefaultDriverProvider.newInstance(DefaultDriverProvider.java:53)
  at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:54)
  at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:214)
  at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
  at java.util.concurrent.FutureTask.run(FutureTask.java:262)
  at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:168)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
  at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
  at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:59)
  ... 9 more
Caused by: org.openqa.selenium.remote.UnreachableBrowserException: Failed to connect to SafariDriver after 10057 ms
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'L-SB80X4FFT4-M.local', ip: '192.168.2.34', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_60'
Driver info: driver.version: SafariDriver
  at org.openqa.selenium.safari.SafariDriverCommandExecutor.start(SafariDriverCommandExecutor.java:121)
  at org.openqa.selenium.safari.SafariDriver.startClient(SafariDriver.java:72)
  at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:114)
  at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:139)
  at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:58)
  at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:51)
  ... 14 more
[SERVER] [stderr] 05:49:47.056,- Exception: Failed to connect to SafariDriver after 10057 ms
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'L-SB80X4FFT4-M.local', ip: '192.168.2.34', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_60'
Driver info: driver.version: SafariDriver
  1) "before all" hook
[SERVER] [stderr] 05:49:47.087 INFO - Shutting down...

  0 passing (11s)
  1 failing

  1)  "before all" hook:
     Error: [init({"browserName":"safari"})] The environment you requested was unavailable.
      at /Users/rroemer/scm/fmd/rowdy/node_modules/wd/lib/webdriver.js:124:15
      at Request._callback (/Users/rroemer/scm/fmd/rowdy/node_modules/wd/lib/http-utils.js:84:7)
      at Request.self.callback (/Users/rroemer/scm/fmd/rowdy/node_modules/wd/node_modules/request/request.js:236:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/Users/rroemer/scm/fmd/rowdy/node_modules/wd/node_modules/request/request.js:1142:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/Users/rroemer/scm/fmd/rowdy/node_modules/wd/node_modules/request/request.js:1096:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:944:16
      at process._tickCallback (node.js:448:13)



npm ERR! Test failed.  See above for more details.
```

--- a/README.md
+++ b/README.md
@@ -46,12 +46,11 @@ Some other tips and tricks:
 
 See https://code.google.com/p/selenium/wiki/InternetExplorerDriver#Required_Configuration
 for extra steps needed to work with local IE. Anecdotally, for Win7+IE11 in a
-VM, the following was needed:
+VM, the only actually tweak needed was:
 
-* "Enable Protected Mode" toggled the same for all zones.
-* (IE11-specific) Created key
+* Create key
   `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Internet Explorer\MAIN\FeatureControl\FEATURE_BFCACHE`
-  and then added DWORD of `iexplore.exe` defaulted to `0` value.
+  and add DWORD of `iexplore.exe` defaulted to `0` value.
 
 ## API
 

--- a/README.md
+++ b/README.md
@@ -42,6 +42,17 @@ $ node_modules/.bin/selenium-standalone install
 
 Some other tips and tricks:
 
+### Mac + Safari
+
+Recent Safari updates have broken automatic driver usage, so you need a series
+of tedious steps per http://elementalselenium.com/tips/69-safari:
+
+* Download the driver at your current version from `https://github.com/SeleniumHQ/selenium/raw/selenium-VERSION/javascript/safari-driver/prebuilt/SafariDriver.safariextz`. A current version used in `selenium-standalone` is `2.45.0`, so
+that would correspond to:
+[ithub.com/SeleniumHQ/selenium/raw/selenium-2.45.0/javascript/safari-driver/prebuilt/SafariDriver.safariextz](https://github.com/SeleniumHQ/selenium/raw/selenium-2.45.0/javascript/safari-driver/prebuilt/SafariDriver.safariextz)
+* Double click the downloaded driver file. Click "Install" when prompted by
+  Safari.
+
 ### Windows + IE
 
 See https://code.google.com/p/selenium/wiki/InternetExplorerDriver#Required_Configuration

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ Main features:
 * Integration helpers for Mocha tests.
 * Use your own test framework and test runner.
 
-## Usage
+## Basic Usage
 
 First, install the library.
 
@@ -31,9 +31,31 @@ var config = require("./PATH/TO/config");
 rowdy(config);
 ```
 
+## Local Usage
+
+One of the Rowdy peer dependencies is `selenium-standalone`. This package
+contains an install script that you may need (and do need on Windows):
+
+```
+$ node_modules/.bin/selenium-standalone install
+```
+
+Some other tips and tricks:
+
+### Windows + IE
+
+See https://code.google.com/p/selenium/wiki/InternetExplorerDriver#Required_Configuration
+for extra steps needed to work with local IE. Anecdotally, for Win7+IE11 in a
+VM, the following was needed:
+
+* "Enable Protected Mode" toggled the same for all zones.
+* (IE11-specific) Created key
+  `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Internet Explorer\MAIN\FeatureControl\FEATURE_BFCACHE`
+  and then added DWORD of `iexplore.exe` defaulted to `0` value.
+
 ## API
 
-*TODO*
+**TODO**: API is still settling out.
 
 ### Examples:
 

--- a/examples/mocha/config.js
+++ b/examples/mocha/config.js
@@ -1,7 +1,10 @@
 /**
  * Configurations.
  */
+// Windows: Phantom path inference.
 var path = require("path");
+var PHANTOM_PATH = process.platform === "win32" && path.join(__dirname,
+  "../../node_modules/phantomjs/lib/phantom/phantomjs.exe");
 
 // Travis
 var BUILD = process.env.TRAVIS_BUILD_NUMBER ?
@@ -59,17 +62,13 @@ module.exports = {
           browserName: "phantomjs"
         },
         server: {
-          start: true
+          start: true,
+          phantomPath: PHANTOM_PATH
         }
       },
       phantomjs: {
         desiredCapabilities: {
           browserName: "phantomjs"
-        },
-        server: {
-          // Add path to NPM-installed Phantom EXE on Windows.
-          phantomPath: process.platform === "win32" && path.join(__dirname,
-            "../../node_modules/phantomjs/lib/phantom/phantomjs.exe")
         }
       },
       firefox: {

--- a/examples/mocha/config.js
+++ b/examples/mocha/config.js
@@ -1,6 +1,8 @@
 /**
  * Configurations.
  */
+var path = require("path");
+
 // Travis
 var BUILD = process.env.TRAVIS_BUILD_NUMBER ?
   process.env.TRAVIS_BUILD_NUMBER + "@" + process.env.TRAVIS_COMMIT :
@@ -56,16 +58,18 @@ module.exports = {
         desiredCapabilities: {
           browserName: "phantomjs"
         },
-        selenium: {
-          // Start local server?
-          start: true,
-          // Path to PhantomJS binary (only set if need to override).
-          phantomPath: false
+        server: {
+          start: true
         }
       },
       phantomjs: {
         desiredCapabilities: {
           browserName: "phantomjs"
+        },
+        server: {
+          // Add path to NPM-installed Phantom EXE on Windows.
+          phantomPath: process.platform === "win32" && path.join(__dirname,
+            "../../node_modules/phantomjs/lib/phantom/phantomjs.exe")
         }
       },
       firefox: {

--- a/examples/mocha/config.js
+++ b/examples/mocha/config.js
@@ -27,7 +27,8 @@ module.exports = {
    */
   options: {
     clientLogger: false,
-    serverLogger: false
+    serverLogger: false,
+    serverDebug: false
   },
 
   /**
@@ -55,7 +56,12 @@ module.exports = {
         desiredCapabilities: {
           browserName: "phantomjs"
         },
-        startLocal: true
+        selenium: {
+          // Start local server?
+          start: true,
+          // Path to PhantomJS binary (only set if need to override).
+          phantomPath: false
+        }
       },
       phantomjs: {
         desiredCapabilities: {

--- a/index.js
+++ b/index.js
@@ -33,7 +33,7 @@ Object.defineProperty(rowdy, "config", {
  */
 Object.defineProperty(rowdy, "setting", {
   get: function () {
-    return rowdy.config._setting;
+    return rowdy.config.setting;
   }
 });
 
@@ -46,7 +46,7 @@ rowdy.setupServer = function (callback) {
   var server = new Server(rowdy.config);
 
   // Start selenium and wait until ready.
-  if (rowdy.setting.selenium.start) {
+  if (rowdy.setting.server.start) {
     return server.start(function (err) {
       callback(err, server);
     });
@@ -77,7 +77,7 @@ rowdy.setupClient = function (callback) {
  * Tear down Selenium server and other state.
  */
 rowdy.teardownServer = function (server, callback) {
-  if (rowdy.setting.selenium.start) {
+  if (rowdy.setting.server.start) {
     return server.stop(callback);
   }
 

--- a/index.js
+++ b/index.js
@@ -46,7 +46,7 @@ rowdy.setupServer = function (callback) {
   var server = new Server(rowdy.config);
 
   // Start selenium and wait until ready.
-  if (rowdy.setting.startLocal) {
+  if (rowdy.setting.selenium.start) {
     return server.start(function (err) {
       callback(err, server);
     });
@@ -77,7 +77,7 @@ rowdy.setupClient = function (callback) {
  * Tear down Selenium server and other state.
  */
 rowdy.teardownServer = function (server, callback) {
-  if (rowdy.setting.startLocal) {
+  if (rowdy.setting.selenium.start) {
     return server.stop(callback);
   }
 

--- a/index.js
+++ b/index.js
@@ -46,7 +46,7 @@ rowdy.setupServer = function (callback) {
   var server = new Server(rowdy.config);
 
   // Start selenium and wait until ready.
-  if (rowdy.setting.server.start) {
+  if ((rowdy.setting.server || {}).start) {
     return server.start(function (err) {
       callback(err, server);
     });
@@ -77,7 +77,7 @@ rowdy.setupClient = function (callback) {
  * Tear down Selenium server and other state.
  */
 rowdy.teardownServer = function (server, callback) {
-  if (rowdy.setting.server.start) {
+  if ((rowdy.setting.server || {}).start) {
     return server.stop(callback);
   }
 

--- a/lib/config.js
+++ b/lib/config.js
@@ -45,7 +45,7 @@ var DEFAULT_SETTING = {
   desiredCapabilities: {
     browserName: "phantomjs",
   },
-  selenium: {
+  server: {
     start: true,
     phantomPath: false
   }
@@ -97,9 +97,9 @@ module.exports = function (cfg) {
     });
   });
 
-  // Finally, choose desired configuration and place on `_setting`.
+  // Finally, choose desired configuration and place on `setting`.
   var settingsPath = ROWDY_SETTINGS.replace(QUOTES_RE, "").split(".");
-  cfg._setting = _.reduce(settingsPath, function (memo, key) {
+  cfg.setting = _.reduce(settingsPath, function (memo, key) {
     if (!_.has(memo, key)) {
       throw new Error("Bad ROWDY_SETTINGS path: " + ROWDY_SETTINGS);
     }

--- a/lib/config.js
+++ b/lib/config.js
@@ -34,6 +34,7 @@ var _ = require("lodash");
 
 var ROWDY_SETTINGS = process.env.ROWDY_SETTINGS || "default";
 var ROWDY_OPTIONS = process.env.ROWDY_OPTIONS || "{}";
+var QUOTES_RE = /^\s*[\\\'\"]+\s*|\s*[\\\'\"]+\s*$/g;
 
 /**
  * Global default: local phantomjs.
@@ -72,7 +73,7 @@ module.exports = function (cfg) {
 
   // Merge options.
   try {
-    _.merge(cfg.options, JSON.parse(ROWDY_OPTIONS));
+    _.merge(cfg.options, JSON.parse(ROWDY_OPTIONS.replace(QUOTES_RE, "")));
   } catch (err) {
     console.error("Unable to parse ROWDY_OPTIONS: " + err.toString());
     throw err;
@@ -94,7 +95,7 @@ module.exports = function (cfg) {
   });
 
   // Finally, choose desired configuration and place on `_setting`.
-  var settingsPath = ROWDY_SETTINGS.split(".");
+  var settingsPath = ROWDY_SETTINGS.replace(QUOTES_RE, "").split(".");
   cfg._setting = _.reduce(settingsPath, function (memo, key) {
     if (!_.has(memo, key)) {
       throw new Error("Bad ROWDY_SETTINGS path: " + ROWDY_SETTINGS);

--- a/lib/config.js
+++ b/lib/config.js
@@ -34,7 +34,7 @@ var _ = require("lodash");
 
 var ROWDY_SETTINGS = process.env.ROWDY_SETTINGS || "default";
 var ROWDY_OPTIONS = process.env.ROWDY_OPTIONS || "{}";
-var QUOTES_RE = /^\s*[\\\'\"]+\s*|\s*[\\\'\"]+\s*$/g;
+var QUOTES_RE = /^[\s\\\'\"]*|[\s\\\'\"]*$/g;
 
 /**
  * Global default: local phantomjs.

--- a/lib/config.js
+++ b/lib/config.js
@@ -45,7 +45,10 @@ var DEFAULT_SETTING = {
   desiredCapabilities: {
     browserName: "phantomjs",
   },
-  startLocal: true
+  selenium: {
+    start: true,
+    phantomPath: false
+  }
 };
 
 var DEFAULTS = {

--- a/lib/config.js
+++ b/lib/config.js
@@ -32,7 +32,7 @@
  */
 var _ = require("lodash");
 
-var ROWDY_SETTINGS = process.env.ROWDY_SETTINGS || "default";
+var ROWDY_SETTINGS = process.env.ROWDY_SETTINGS || "";
 var ROWDY_OPTIONS = process.env.ROWDY_OPTIONS || "{}";
 var QUOTES_RE = /^[\s\\\'\"]*|[\s\\\'\"]*$/g;
 
@@ -41,7 +41,7 @@ var QUOTES_RE = /^[\s\\\'\"]*|[\s\\\'\"]*$/g;
  *
  * Note: Only if *no* default top-level setting specified.
  */
-var DEFAULT_SETTING = {
+var GLOBAL_DEFAULT = {
   desiredCapabilities: {
     browserName: "phantomjs",
   },
@@ -82,9 +82,9 @@ module.exports = function (cfg) {
     throw err;
   }
 
-  // Apply default setting if not set.
+  // Add default setting if not set.
   if (!cfg.settings.default) {
-    cfg.settings.default = DEFAULT_SETTING;
+    cfg.settings.default = GLOBAL_DEFAULT;
   }
 
   // Then, apply merge settings defaults to other objects one level down.
@@ -99,6 +99,20 @@ module.exports = function (cfg) {
 
   // Finally, choose desired configuration and place on `setting`.
   var settingsPath = ROWDY_SETTINGS.replace(QUOTES_RE, "").split(".");
+  settingsPath = _.chain(settingsPath)
+    .map(function (v) { return v.trim(); })
+    .filter(_.identity)
+    .value();
+
+  // If no settings path, _try_ to go for `local.default`, then `default`.
+  if (!settingsPath.length) {
+    if (cfg.settings.local && cfg.settings.local.default) {
+      settingsPath = ["local", "default"];
+    } else {
+      settingsPath = ["default"];
+    }
+  }
+
   cfg.setting = _.reduce(settingsPath, function (memo, key) {
     if (!_.has(memo, key)) {
       throw new Error("Bad ROWDY_SETTINGS path: " + ROWDY_SETTINGS);

--- a/lib/server.js
+++ b/lib/server.js
@@ -23,7 +23,6 @@ var STOP_MAX_WAIT_MS = 5 * 1000;
 
 var Server = module.exports = function (config) {
   this.config = config || {};
-  this.setting = this.config._setting || {};
   this._process = null;
   this._isStarted = false;
   this._err = null;
@@ -71,7 +70,7 @@ Server.prototype.start = function (callback) {
   var self = this;
   var doLog = self.config.options.serverLogger;
   var debug = self.config.options.serverDebug;
-  var phantomPath = self.setting.selenium.phantomPath;
+  var phantomPath = self.config.setting.server.phantomPath;
   var timerId;
 
   // Verify and clean state.

--- a/lib/server.js
+++ b/lib/server.js
@@ -23,6 +23,7 @@ var STOP_MAX_WAIT_MS = 5 * 1000;
 
 var Server = module.exports = function (config) {
   this.config = config || {};
+  this.setting = this.config._setting || {};
   this._process = null;
   this._isStarted = false;
   this._err = null;
@@ -70,6 +71,7 @@ Server.prototype.start = function (callback) {
   var self = this;
   var doLog = self.config.options.serverLogger;
   var debug = self.config.options.serverDebug;
+  var phantomPath = self.setting.selenium.phantomPath;
   var timerId;
 
   // Verify and clean state.
@@ -86,7 +88,10 @@ Server.prototype.start = function (callback) {
 
   // Start selenium process.
   selenium.start({
-    seleniumArgs: debug ? ["-debug"] : [],
+    seleniumArgs: [].concat(
+      debug ? ["-debug"] : [],
+      phantomPath ? ["-Dphantomjs.binary.path=" + phantomPath] : []
+    ),
     spawnOptions: {
       stdio: "pipe"
     },

--- a/lib/server.js
+++ b/lib/server.js
@@ -10,7 +10,8 @@ var _log = log.bind(log, "[SERVER]".yellow);
 var STARTED = "Started org.openqa.jetty.jetty.Server";
 var ERRORS = {
   ALREADY_RUNNING: "java.net.BindException: Selenium is already running",
-  FAILED_TO_START: "Failed to start: SocketListener"
+  FAILED_TO_START: "Failed to start: SocketListener",
+  EXECUTE_EXCEPTION: "org.apache.commons.exec.ExecuteException"
 };
 var SEL_LOG_COLORS = {
   " DEBUG ": "blue",

--- a/package.json
+++ b/package.json
@@ -18,7 +18,7 @@
     "mocha": "1.21.4",
     "phantomjs": "1.9.9",
     "selenium-standalone": "4.2.2",
-    "wd": "0.3.8"
+    "wd": "0.3.11"
   },
   "repository": {
     "type": "git",

